pulsar-broker/src/main/java/org/apache/pulsar/broker/namespace/NamespaceService.java

```diff
@@ -596,7 +599,8 @@ private void searchForCandidateBroker(NamespaceBundle bundle,
                 }
 
                 // Now setting the redirect url
-                createLookupResult(candidateBroker, authoritativeRedirect, options.getAdvertisedListenerName())
+                createLookupResult(candidateBrokerAdvertisedAddr == null ? candidateBroker
```

Comment:
When I debugged `testLookup`, it didn't go to the `else` block so that the `candidateBrokerAdvertisedAddr` doesn't take affect.

<img width="1015" alt="image" src="https://user-images.githubusercontent.com/18204803/168951432-9764d1bf-2057-411f-be47-e805351355ab.png">

Reply:
Sorry there's something wrong with my debugging, mark it as resolved.